packages/flutter/lib/src/material/chip.dart

```diff
@@ -2049,7 +2051,9 @@ class _RenderChip extends RenderBox with SlottedContainerRenderObjectMixin<_Chip
     final Color disabledColor = _disabledColor;
     final int disabledColorAlpha = disabledColor.alpha;
     if (needsCompositing) {
-      context.pushLayer(OpacityLayer(alpha: disabledColorAlpha), paintWithOverlay, offset);
+      final OpacityLayer newLayer = OpacityLayer(alpha: disabledColorAlpha);
+      _layerHandle.layer = newLayer;
+      context.pushLayer(newLayer, paintWithOverlay, offset);
```

Comment:
Can this be simplified to:

```suggestion
      _layerHandle.layer = context.pushOpacity(offset, disabledColorAlpha, paintWithOverlay, oldLayer: _layerHandle.layer);      
```

Reply:
Additionally, you want to null out `_layerHandle.layer` in the code paths that don't re-use the layer, that is in the early return in line 2049 and in the else branch of this if right here.